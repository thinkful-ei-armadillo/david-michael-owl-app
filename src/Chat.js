import React from 'react';
import Store from './STORE'

function Chat (){
  const { participants, chatEvents} = Store;

   return (
     <div className="chatBox">
     {chatEvents.map((event, index) => {
       let id = event.participantId;
       let name = participants.find((person) => person.id === id).name;
       let message = event.message
       let time = new Date(event.timestamp).toString()
        if(event.type === 'message'){
          return (
            <div className="chatEvent" key ={index}>
            <p className="name" >

              {name}: {message}
            </p>
         </div>
          )
       } else if (event.type === 'thumbs-up'){
         return (
          <div className="chatEvent" key ={index}>
          <p className="message">
            {time}
            {name} gave a thumbs-up
          </p>
       </div>
         )
       } else if (event.type === 'thumbs-down'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} gave a thumbs-down
          </p>
       </div>
         )
       } else if (event.type === 'clap'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} clapped
          </p>
       </div>
         )
       } else if (event.type === 'raise-hand'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} raised their hand
          </p>
       </div>
         )
       } else if(event.type === 'clap'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} clapped
          </p>
       </div>
         )
       } else if (event.type === 'join'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} joined
          </p>
       </div>
         )
       } else if (event.type === 'join-stage'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} joined the stage
          </p>
       </div>
         )
       } else if(event.type === 'leave-stage'){
        return (
          <div className="chatEvent" key ={index}>
          <p className="message">
          {time}
            {name} left the stage
          </p>
       </div>
         )
       }
      


       
     })}

      
      </div>
   )

}

export default Chat;


