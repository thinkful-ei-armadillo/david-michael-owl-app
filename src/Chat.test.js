import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Chat from './Chat'

describe('<Chat />', () => {
    it('renders without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<Chat type='string' timestamp={2} name='steve' avatar = 'image string'  />, div);
        ReactDOM.unmountComponentAtNode(div);
      })
    })



describe('<Chat />', () => {
  it('renders without crashing', () => {
      // Render the component, as JSON
      const tree = renderer.create(<Chat type='string' timestamp={2} name='steve' avatar = 'image string'  />).toJSON();
      // Check whether it matches the previous snapshot
      // Stored in __snapshots__/App.test.js.snap
      expect(tree).toMatchSnapshot(); 
  });
});
