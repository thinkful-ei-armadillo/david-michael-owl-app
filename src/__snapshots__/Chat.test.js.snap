// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders without crashing 1`] = `
<div
  className="chatBox"
>
  <div
    className="chatEvent"
  >
    <p
      className="name"
    >
      Axe Kubicka
      : 
      Hello world
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:48:04 GMT-0600 (CST)
      Caty Flucker
       gave a thumbs-up
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:48:04 GMT-0600 (CST)
      Frank Runciman
       gave a thumbs-down
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:49:04 GMT-0600 (CST)
      Koren Templeton
       raised their hand
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:49:04 GMT-0600 (CST)
      Koren Templeton
       clapped
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:49:04 GMT-0600 (CST)
      Koren Templeton
       joined
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:51:04 GMT-0600 (CST)
      Axe Kubicka
       joined the stage
    </p>
  </div>
  <div
    className="chatEvent"
  >
    <p
      className="message"
    >
      Wed Jan 30 2019 06:52:04 GMT-0600 (CST)
      Axe Kubicka
       left the stage
    </p>
  </div>
</div>
`;
